pennylane/workflow/qnode.py

```diff
@@ -754,10 +766,48 @@ def circuit(x):
 
         original_init_args.update(kwargs)
         updated_qn = QNode(**original_init_args)
+
+        # If device was updated, shots would have been updated
+        # but the the `shots` arg should override if exists
+        if shots_to_update:
+            updated_qn._shots = shots
+
         # pylint: disable=protected-access
         updated_qn._transform_program = qml.transforms.core.TransformProgram(self.transform_program)
         return updated_qn
 
+    def update_shots(self, shots: Union[int, qml.measurements.Shots]) -> "QNode":
+        """Update the number of shots used by the QNode.
+
+        Args:
+            shots (int or qml.measurements.Shots): The new number of shots to use.
+
+        Returns:
+            qnode (QNode): new QNode with updated shots
+        """
+        if isinstance(shots, int):
+            shots = qml.measurements.Shots(shots)
+
+        # Create a copy of the current QNode
+        updated_qn = copy.copy(self)
+
+        # Update the shots attribute directly
+        updated_qn._set_shots(shots)
+
+        return updated_qn
+
+    def _set_shots(self, shots: Union[int, qml.measurements.Shots]) -> None:
+        """Set the number of shots used by the QNode.
+
+        Args:
+            shots (int or qml.measurements.Shots): The new number of shots to use.
+        """
+        if isinstance(shots, int):
+            shots = qml.measurements.Shots(shots)
```

Comment:
Why only for ints? I feel like we should enforce that `self._shots` is always a `Shots` object.